botfront/imports/ui/components/stories/common/UserUtterancesContainer.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import IconButton from '../../common/IconButton';
+import UserUtteranceContainer from './UserUtteranceContainer';
+import { NEW_INTENT } from '../../../../lib/story_controller';
+
+const UserUtterancesContainer = (props) => {
+    const {
+        deletable, value, onChange, onDelete,
+    } = props;
+
+    const somethingIsBeingInput = useMemo(() => value.some(d => d === null), [value]);
+
+    const handleDeleteDisjunct = (index) => {
+        const newUtterance = [...value.slice(0, index), ...value.slice(index + 1)];
+        if (newUtterance.length) {
+            return onChange([...value.slice(0, index), ...value.slice(index + 1)]);
```

Comment:
Minor: maybe just remplace the second `[...value.slice(0, index), ...value.slice(index + 1)]`by `newUtterance` ? :)